standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -0,0 +1,1641 @@
+   * @return value, or default value if value not in config file
+   */
+  public static int getIntVar(Configuration conf, ConfVars var) {
+    long val = getLongVar(conf, var);
+    assert val <= Integer.MAX_VALUE;
+    return (int)val;
+  }
+
+  /**
+   * Get the variable as a long
+   * @param conf configuration to retrieve it from
+   * @param var variable to retrieve
+   * @return value, or default value if value not in config file
+   */
+  public static long getLongVar(Configuration conf, ConfVars var) {
+    assert var.defaultVal.getClass() == Long.class;
+    String val = conf.get(var.varname);
+    return val == null ? conf.getLong(var.hiveName, (Long)var.defaultVal) : Long.valueOf(val);
+  }
+
+  /**
+   * Set the variable as a long
+   * @param conf configuration file to set it in
+   * @param var variable to set
+   * @param val value to set it to
+   */
+  public static void setLongVar(Configuration conf, ConfVars var, long val) {
+    assert var.defaultVal.getClass() == Long.class;
+    conf.setLong(var.varname, val);
+  }
+
+  /**
+   * Get the variable as a boolean
+   * @param conf configuration to retrieve it from
+   * @param var variable to retrieve
+   * @return value, or default value if value not in config file
+   */
+  public static boolean getBoolVar(Configuration conf, ConfVars var) {
+    assert var.defaultVal.getClass() == Boolean.class;
+    String val = conf.get(var.varname);
+    return val == null ? conf.getBoolean(var.hiveName, (Boolean)var.defaultVal) : Boolean.valueOf(val);
+  }
+
+  /**
+   * Set the variable as a boolean
+   * @param conf configuration file to set it in
+   * @param var variable to set
+   * @param val value to set it to
+   */
+  public static void setBoolVar(Configuration conf, ConfVars var, boolean val) {
+    assert var.defaultVal.getClass() == Boolean.class;
+    conf.setBoolean(var.varname, val);
+  }
+
+  /**
+   * Get the variable as a double
+   * @param conf configuration to retrieve it from
+   * @param var variable to retrieve
+   * @return value, or default value if value not in config file
+   */
+  public static double getDoubleVar(Configuration conf, ConfVars var) {
+    assert var.defaultVal.getClass() == Double.class;
+    String val = conf.get(var.varname);
+    return val == null ? conf.getDouble(var.hiveName, (Double)var.defaultVal) : Double.valueOf(val);
+  }
+
+  /**
+   * Set the variable as a double
+   * @param conf configuration file to set it in
+   * @param var variable to set
+   * @param val value to set it to
+   */
+  public static void setDoubleVar(Configuration conf, ConfVars var, double val) {
+    assert var.defaultVal.getClass() == Double.class;
+    conf.setDouble(var.varname, val);
+  }
+
+  /**
+   * Get a class instance based on a configuration value
+   * @param conf configuration file to retrieve it from
+   * @param var variable to retrieve
+   * @param defaultValue default class to return if the value isn't set
+   * @p
```

Comment:
Would be helpful if we add methods to getMetastoreSiteLocation too. Additionally, I think it will be useful to add a utility method which prints the URLs of all the config file locations which are being used to construct this Configuration object. Eg:
List<URL> getResourceFileLocations() {
   urls.add(getMetastoreSiteLocation());
   urls.add(getMetastoreSiteLocation());
   urls.add(getHiveSiteLocation());
}